frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -0,0 +1,471 @@
+  namespace,
+  csvName,
+  obj,
+}) => {
+  const crdMessage = getCRDFromObject(obj)
+    ? 'The operator has installed successfully. Create the required custom resource to be able to use this operator.'
+    : '';
+  return (
+    <>
+      <h2 className="co-clusterserviceversion-install__heading">
+        Installed operator - ready for use
+      </h2>
+      <span>{crdMessage}</span>
+      <InstallPageCRDRequiredMessage namespace={namespace} obj={obj} />
+      <ActionGroup className="pf-c-form pf-c-form__group--no-top-margin">
+        {!getCRDFromObject(obj) && (
+          <Button
+            variant="primary"
+            component="a"
+            href={resourcePathFromModel(ClusterServiceVersionModel, csvName, namespace)}
+          >
+            View Operator
+          </Button>
+        )}
+        <CreateCRDButton namespace={namespace} obj={obj} />
+        <ViewInstalledOperatorsButton namespace={namespace} />
+      </ActionGroup>
+    </>
+  );
+};
+const InstallingMessage: React.FC<InstallPageMessageProps> = ({ namespace, obj }) => {
+  const reason = (obj as ClusterServiceVersionKind)?.status?.reason || '';
+  const message = (obj as ClusterServiceVersionKind)?.status?.message || '';
+  const crdMessage = getCRDFromObject(obj)
+    ? 'Once the operator is installed the required custom resource will be available for creation.'
+    : '';
+  const installMessage = `The operator is being installed. This may take a few minutes. ${crdMessage}`;
+  return (
+    <>
+      <h2 className="co-clusterserviceversion-install__heading">Installing operator</h2>
+      {reason && (
+        <div className="text-muted">
+          {reason}: {message}
+        </div>
+      )}
+      <span>{installMessage}</span>
+      <InstallPageCRDRequiredMessage namespace={namespace} obj={obj} />
+      <ActionGroup className="pf-c-form pf-c-form__group--no-top-margin">
+        {getCRDFromObject(obj) && (
+          <CreateCRDButton namespace={namespace} className="" obj={obj} disabled />
+        )}
+        <ViewInstalledOperatorsButton
+          namespace={namespace}
+          className={getCRDFromObject(obj) ? '' : 'co-clusterserviceversion__link--align-left'}
+        />
+      </ActionGroup>
+    </>
+  );
+};
+
+const OperatorInstallStatusPage: React.FC<OperatorInstallPageProps> = (props) => {
+  const [loading, setLoading] = React.useState(true);
+  const [installObj, setInstallObj] = React.useState<ClusterServiceVersionKind | InstallPlanKind>();
+  const [subscription, setSubscription] = React.useState<SubscriptionKind>();
+  const [status, setStatus] = React.useState('');
+
+  const { resources, targetNamespace, pkgNameWithVersion } = props;
+
+  const isStatusSucceeded = status === 'Succeeded';
+  const isStatusFailed = status === 'Failed';
+  const isApprovalNeeded =
+    installObj?.spec?.approval === 'Manual' && installObj?.spec?.approved === false;
+
+  React.useEffect(() => {
```

Comment:
`useEffect` and `setState` still seem wrong here. Just calculate the value and set it to a local var. If you're concerned about performance, you can always use `useMemo`, but I don't think it's necessary.